crates/cxx-qt-gen/src/generator/rust/property/getter.rs

```diff
@@ -32,14 +38,14 @@ pub fn generate(
         implementation: vec![
             quote! {
                 impl #rust_struct_name_rust {
-                    pub fn #getter_rust<'a>(&'a self, cpp: &'a #cpp_class_name_rust) -> &'a #ty {
+                    pub #has_unsafe fn #getter_rust<'a>(&'a self, cpp: &'a #cpp_class_name_rust) -> &'a #ty {
                         cpp.#getter_rust()
                     }
                 }
             },
             quote! {
                 impl #cpp_class_name_rust {
-                    pub fn #getter_rust(&self) -> &#ty {
+                    pub #has_unsafe fn #getter_rust(&self) -> &#ty {
```

Comment:
C++ getter ends up discarding qualifiers

```
[build]   cargo:warning=/workspaces/cxx-qt/build/debug-ninja-sscache/./cargo/build/x86_64-unknown-linux-gnu/debug/build/qml-features-237b4fae8b31c37a/out/cxx-qt-gen/src/nested_qobjects.cxx.cpp: In member function ‘InnerObject* const& OuterObjectRust::getInner(const OuterObject&) const’:
[build]   cargo:warning=/workspaces/cxx-qt/build/debug-ninja-sscache/./cargo/build/x86_64-unknown-linux-gnu/debug/build/qml-features-237b4fae8b31c37a/out/cxx-qt-gen/src/nested_qobjects.cxx.cpp:576:10: error: binding reference of type ‘InnerObject* const&’ to ‘const InnerObject*’ discards qualifiers
```

Reply:
This seems to be coming from the CXX generated method

```cpp
const ::InnerObject * *cxxbridge1$OuterObjectRust$inner(const ::OuterObjectRust &self, const ::OuterObject &cpp) noexcept;

::InnerObject *const &OuterObjectRust::getInner(const ::OuterObject &cpp) const noexcept {
  return *cxxbridge1$OuterObjectRust$inner(*this, cpp);
}
```

Reply:
By just returning `*mut T` this then works, just means we have a special case for the getter in C++ and Rust.

Reply:
Solved upstream.